### DISCUSSION TITLE
Update order "Cancel Items" UI/ UX

### DISCUSSION THREAD

**Discussion by mdamian322** [solidusio/solidus#4579] [New Features or Ideas]
**Is your feature request related to a problem? Please describe.**

When you need to cancel a line item that is like 100 quantity (think bookmarks), you have to select a checkbox 100 times to cancel all of them. The view gets very long (see last screenshot). And if you had to cancel product1 with quantity 1 but you also have product2 with quantity 20 and product1 happens to be at the bottom, that can be a lot of scrolling. Imagine those scenarios with more than 2 products in the order.

**Describe the solution you'd like**

- It would be cleaner to group the inventory units by line item and state so there are not as many rows and the user can cancel backordered ones first.
- Instead of a checkbox we have a dropdown that is default `0` and the max number in the dropdown is the quantity of the row (example, if quantity is 5, the numbers in dropdown go up to 5)
- The max of the number for the dropdown is 10 to prevent a long select list.
- If the row quantity is greater than 10, we show 10+ as last option.
- When 10+ is selected dropdown changes to an input box with the save icon on the side (checkmark). After clicking the save icon, it remains an input, not a dropdown, until the amount is set to less than 10, which will change back to a dropdown.
- Clicking in and out of the input box will toggle the save icon. If you click outside of the input without clicking the icon, changes are not saved.
 
**Describe alternatives you've considered**

We thought about having a "Select All" button next to "Cancel Items" and then they could deselect what they don't want to cancel. But if they have an order that has 10 of product1 and 10 of product2, either way they have to click a checkbox 10 times even with the “select all” button. The issue is still there, just backwards.

**Additional context**

**Dropdown with max of row quantity (up to 10)**

<img width="1290" alt="dropdown-max-is-quantity" src="https://user-images.githubusercontent.com/13877138/88322251-ab2e5300-ccdd-11ea-8b96-b722f4b92bd3.png">

<img width="1292" alt="Screen Shot 2020-07-23 at 10 57 56 AM" src="https://user-images.githubusercontent.com/13877138/88322471-019b9180-ccde-11ea-9c90-ecfa9dcd0ef5.png">

**Dropdown when quantity is greater than 10, show 10+ option**

<img width="1290" alt="10+-dropdown" src="https://user-images.githubusercontent.com/13877138/88322375-ddd84b80-ccdd-11ea-823b-aae7efa38c4c.png">

**After selecting 10+**

<img width="897" alt="input-with-save-icon" src="https://user-images.githubusercontent.com/13877138/88322584-27c13180-ccde-11ea-96fa-0ed592a768ab.png">

**After clicking the save icon, or when the input remains greater than 10 (static input state)**

<img width="1293" alt="input-state-when-greater-than-10" src="https://user-images.githubusercontent.com/13877138/88322677-49221d80-ccde-11ea-907d-a88c25e746c7.png">

**Current view example with 100 of product1 and 1 of product2**
![screencapture-localhost-3000-admin-orders-R392242979-cancellations-1595529451613](https://user-images.githubusercontent.com/13877138/88324751-5db3e500-cce1-11ea-9548-82b1ff4dc0ba.png)

**Comment by jarednorman** [comment]
Thanks for this issue report. This definitely looks like an area that needs improvement.
